packages/adapters/wagmi/src/connectors/UniversalConnector.ts

```diff
@@ -209,28 +208,8 @@ export function walletConnect(
       return accounts as `0x${string}`[]
     },
     async getProvider({ chainId } = {}) {
-      async function initProvider() {
-        const optionalChains = caipNetworks.map(x => Number(x.id))
-
-        if (!optionalChains.length) {
-          return undefined
-        }
-
-        const provider = await appKit.getUniversalProvider()
-
-        if (!provider) {
-          throw new Error('Provider not found')
-        }
-
-        return provider
-      }
-
       if (!provider_) {
-        if (!providerPromise) {
-          providerPromise = initProvider()
-        }
-        provider_ = await providerPromise
-        provider_?.events.setMaxListeners(Number.POSITIVE_INFINITY)
```

Comment:
why remove this and why was it there in the first place?
cc @svenvoskamp ?

Reply:
This issue was happening because when connecting on refresh we [would check for stale chains](https://github.com/reown-com/appkit/blob/main/packages/adapters/wagmi/src/connectors/UniversalConnector.ts#L106-L110) without setting it then previously when calling through one click auth `authenticate` of SIWE/SIWX. This would force a non intended disconnect, which was losing the connection in page refresh.

So I've [added in the `onConnect`](https://github.com/reown-com/appkit/pull/3289/files#diff-e2ce97a9a1285d808ba62302910fcc120b2994f58e50cb77e12576c1d8bf1b0aR351) which is the callback of `authenticate` to set the requested chains.

This code snippet you tagged was only a code clean up that seems to be leftover from multichain refactor.